### PR TITLE
Add typings

### PR DESCRIPTION
**Pull request by OleksiiKachan**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

**Comment by OleksiiKachan** [comment]
@tshemsedinov please review

**Comment by OleksiiKachan** [comment]
after this one is merged, gonna merge metarhia:master into my upsert branch and will add upsert function to declarations there

**Comment by OleksiiKachan** [comment]
https://github.com/metarhia/sql/issues/17

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## [Unreleased][unreleased]
 
 - Support `OR` in queries
+- Add typing for Database and Query classes
```
Comment (tshemsedinov) [inline CHANGELOG.md:6]:
```suggestion
- Add typings for Database and Query classes
```

File: types/metasql.d.ts [hunk 2]
```diff
@@ -1 +1,62 @@
...
+  row(
+    table: string,
+    fields: Array<string>,
+    ...conditions: Array<object>
+  ): Promise<Array<object>>;
+  scalar(
+    table: string,
+    field: string,
+    ...conditions: Array<object>
+  ): Promise<string | number | undefined>;
```
Comment (tshemsedinov) [inline types/metasql.d.ts:30]: I'd better define type `ScalarValue` to reuse declaration

File: types/metasql.d.ts [hunk 3]
```diff
@@ -1 +1,62 @@
...
+  col: (
+    table: string,
+    field: string,
+    ...conditions: Array<object>
+  ) => Promise<Array<string | number | undefined>>;
```
Comment (tshemsedinov) [inline types/metasql.d.ts:35]:
```suggestion
  ): Promise<Array<string | number | undefined>>;
```

File: types/metasql.d.ts [hunk 4]
```diff
@@ -1 +1,62 @@
...
+  dict: (
+    table: string,
+    fields: Array<string>,
+    ...conditions: Array<object>
+  ) => Promise<object>;
```
Comment (tshemsedinov) [inline types/metasql.d.ts:40]:
```suggestion
  ): Promise<object>;
```

File: types/metasql.d.ts [hunk 5]
```diff
@@ -1 +1,62 @@
...
+  delete: (table: string, ...conditions: Array<object>) => Promise<QueryResult>;
```
Comment (tshemsedinov) [inline types/metasql.d.ts:41]:
```suggestion
  delete(table: string, ...conditions: Array<object>): Promise<QueryResult>;
```

File: types/metasql.d.ts [hunk 6]
```diff
@@ -1 +1,62 @@
...
+  update: (
+    table: string,
+    delta: object,
+    ...conditions: Array<object>
+  ) => Promise<QueryResult>;
```
Comment (tshemsedinov) [inline types/metasql.d.ts:46]:
```suggestion
  ): Promise<QueryResult>;
```

File: types/metasql.d.ts [hunk 7]
```diff
@@ -1 +1,62 @@
...
+  close: () => void;
```
Comment (tshemsedinov) [inline types/metasql.d.ts:47]:
```suggestion
  close(): void;
```

File: types/metasql.d.ts [hunk 8]
```diff
@@ -1 +1,62 @@
...
+    field: string,
+    ...conditions: Array<object>
+  ) => Promise<Array<string | number | undefined>>;
+  dict: (
+    table: string,
+    fields: Array<string>,
+    ...conditions: Array<object>
+  ) => Promise<object>;
+  delete: (table: string, ...conditions: Array<object>) => Promise<QueryResult>;
+  update: (
```
Comment (tshemsedinov) [inline types/metasql.d.ts:42]:
```suggestion
  update(
```

File: types/metasql.d.ts [hunk 9]
```diff
@@ -1 +1,62 @@
...
+    table: string,
+    field: string,
+    ...conditions: Array<object>
+  ): Promise<string | number | undefined>;
+  col: (
+    table: string,
+    field: string,
+    ...conditions: Array<object>
+  ) => Promise<Array<string | number | undefined>>;
+  dict: (
```
Comment (tshemsedinov) [inline types/metasql.d.ts:36]:
```suggestion
  dict(
```

File: types/metasql.d.ts [hunk 10]
```diff
@@ -1 +1,62 @@
...
+    table: string,
+    fields: Array<string>,
+    ...conditions: Array<object>
+  ): Promise<Array<object>>;
+  scalar(
+    table: string,
+    field: string,
+    ...conditions: Array<object>
+  ): Promise<string | number | undefined>;
+  col: (
```
Comment (tshemsedinov) [inline types/metasql.d.ts:31]:
```suggestion
  col(
```

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/metasql/commit/cdcf9e15b3413586140dc07fc3d48896f6260c4d

**Pull request closed without merging**